lib/chef/formatters/error_inspectors/resource_failure_inspector.rb

```diff
@@ -63,7 +63,7 @@ def add_explanation(error_description)
         def recipe_snippet
           return nil if dynamic_resource?
           @snippet ||= begin
-            if file = resource.source_line[/^(([\w]:)?[^:]+):([\d]+)/,1] and line = resource.source_line[/^#{file}:([\d]+)/,1].to_i
+            if file = resource.source_line[/^(([\w]:)?[^:]+):([\d]+)/,1] and line = resource.source_line[/^#{Regexp.escape(file)}:([\d]+)/,1].to_i
```

Comment:
:+1: on the change in general.  If you have some time to clean this up a bit, it might be nice to refactor this into well-named functions.